src/main/kotlin/io/github/moulberry/notenoughupdates/miscfeatures/OldSkyBlockMenu.kt

```diff
@@ -0,0 +1,180 @@
+                        "yNDI3YjI2MGQwYTYxZTY0ODNmYzNiMmMzNWE1ODU4NTFlMDhhOWE5ZGYzNzI1NDhiNDE2OGNjODE3YyJ9fX0="
+            )
+        ),
+        QUIVER(
+            "quiver", 44,
+            "Quiver",
+            "A masterfully crafted Quiver",
+            "which holds any kind of",
+            "projectile you can think of!",
+            itemData = SkullItemData(
+                "41758912-e6b1-4700-9de5-04f2cfb9c422",
+                "eyJ0ZXh0dXJlcyI6eyJTS0lOIjp7InVybCI6Imh0dHA6Ly90ZXh0dXJlcy5taW5lY3JhZnQubmV0L3RleHR1cmUvNGNiM2FjZ" +
+                        "GMxMWNhNzQ3YmY3MTBlNTlmNGM4ZTliM2Q5NDlmZGQzNjRjNjg2OTgzMWNhODc4ZjA3NjNkMTc4NyJ9fX0="
+            )
+        ),
+        FISHING(
+            "fishingbag", 43,
+            "Fishing Bag",
+            "A useful bag which can hold all",
+            "types of fish, baits, and fishing",
+            "loot!",
+            itemData = SkullItemData(
+                "508c01d6-eabe-430b-9811-874691ee7ee4",
+                "eyJ0ZXh0dXJlcyI6eyJTS0lOIjp7InVybCI6Imh0dHA6Ly90ZXh0dXJlcy5taW5lY3JhZnQubmV0L3RleHR1cmUvZWI4ZT" +
+                        "I5N2RmNmI4ZGZmY2YxMzVkYmE4NGVjNzkyZDQyMGFkOGVjYjQ1OGQxNDQyODg1NzJhODQ2MDNiMTYzMSJ9fX0="
+            )
+        ),
+        SACK_OF_SACKS(
+            "sacks", 35,
+            "Sack of Sacks",
+            "A sack which contains other",
+            "sacks. Sackception!",
+            itemData = SkullItemData(
+                "a206a7eb-70fc-4f9f-8316-c3f69d6ba2ca",
+                "ewogICJ0aW1lc3RhbXAiIDogMTU5MTMxMDU4NTYwOSwKICAicHJvZmlsZUlkIiA6ICI0MWQzYWJjMmQ3NDk0MDBjOTA5MGQ1NDM0" +
+                        "ZDAzODMxYiIsCiAgInByb2ZpbGVOYW1lIiA6ICJNZWdha2xvb24iLAogICJzaWduYXR1cmVSZXF1aXJlZCIgOiB0cnVl" +
+                        "LAogICJ0ZXh0dXJlcyIgOiB7CiAgICAiU0tJTiIgOiB7CiAgICAgICJ1cmwiIDogImh0dHA6Ly90ZXh0dXJlcy5taW5l" +
+                        "Y3JhZnQubmV0L3RleHR1cmUvODBhMDc3ZTI0OGQxNDI3NzJlYTgwMDg2NGY4YzU3OGI5ZDM2ODg1YjI5ZGFmODM2YjY0" +
+                        "YTcwNjg4MmI2ZWMxMCIKICAgIH0KICB9Cn0="
+            )
+        ),
+        ;
+
+        fun buildItem(): ItemStack {
+            if (cachedItem != null) return cachedItem!!
+
+            val lore = mutableListOf<String>()
+            for (line in displayDescription) {
+                lore.add("§7$line")
+            }
+            lore.add("")
+            lore.add("§eClick to execute /$command")
+            val array = lore.toTypedArray()
+
+            val name = "§a$displayName"
+            val result = when (itemData) {
+                is NormalItemData -> Utils.createItemStackArray(itemData.displayIcon, name, array)
+                is SkullItemData -> Utils.createSkull(name, itemData.uuid, itemData.value, array)
+
+                else -> {
+                    throw Error("item data error!")
+                }
+            }
+            cachedItem = result
+            return result
+        }
+    }
```

Comment:
Maybe: 
```kt
val item by lazy {}
```

Reply:
need a ted talk for how tpo use lazy

Reply:
It's essentially what you did:

```kt
val someVariable by lazy {/*some expensive function*/} 
// Does not calculate function yet.

// Later...

println(someVariable) // Now expensive function gets called.

// Even Later...

println(someVariable) // Cached Value used

```

its a really simple thing.

in theory you can also specify a thread safety mode (with the default being thread safe = true), but nobody does that. (see `lazy(LazyThreadSafetyMode.PUBLICATION) {}` or similar if you care)

Reply:
It's a bit like `Supplier.memoized` from guava, but with kotlin `val by` syntax easier available and thread safety (optionally) enabled.

Reply:
Oh while we are on the topic of caching: maybe have a lookup table from slotId -> replacment item so you dont have to keep iterating so much.

Reply:
implemented lazy stuff and map stuf and janis hot-loved booster cookie stuff. now im done ig.